pyproj/geod.py

```diff
@@ -336,17 +340,265 @@ def npts(
             Latitude of the terminus point
         npts: int
             Number of points to be returned
+            (including initial and/or terminus points, if required)
         radians: bool, optional
             If True, the input data is assumed to be in radians.
+        initial_idx: int, optional
+            if initial_idx==0 then the initial point would be included in the output
+            (as the first point)
+        terminus_idx: int, optional
+            if terminus_idx==0 then the terminus point would be included in the output
+            (as the last point)
+        return_points: bool, optional
+            determines the return type of the function (see below)
 
         Returns
         -------
-        list of tuples:
-            list of (lon, lat) points along the geodesic
-            between the initial and terminus points.
+        if (return_points == True) (default):
+            list of tuples:
+                list of (lon, lat) points along the geodesic
+                between the initial and terminus points.
+        else
+            tuple of two sequences:
+                tuple of (lons, lats) of the points along the geodesic
+                lons and lats are two seqeunces with the same length
+                between the initial and terminus points.
         """
-        lons, lats = super()._npts(lon1, lat1, lon2, lat2, npts, radians=radians)
-        return list(zip(lons, lats))
+        lons, lats = super()._npts(
+            lon1,
+            lat1,
+            lon2,
+            lat2,
+            npts,
+            radians=radians,
+            initial_idx=initial_idx,
+            terminus_idx=terminus_idx,
+        )
+        return list(zip(lons, lats)) if return_points else (lons, lats)
+
+    def inv_intermediate_by_npts(
+        self,
+        out_lons: Any,
+        out_lats: Any,
+        out_azis: Any,
+        lon1: float,
+        lat1: float,
+        lon2: float,
+        lat2: float,
+        npts: int,
+        initial_idx: int = 1,
+        terminus_idx: int = 1,
+        radians: bool = False,
+    ) -> GeodIntermediateReturn:
+        """
+        Given a single initial point and terminus point,
+        and the number of points, returns
+        a list of longitude/latitude pairs describing npts equally
+        spaced intermediate points along the geodesic between the
+        initial and terminus points.
+        This function calculates the distance between the points by
+        the distance between the initial point and the terminus point
+        divided by npts
+
+        Similar to inv_intermediate_by_npts(),
+        but using npts as input as appose to del_s
+
+        Similar to npts(), but using the given buffers for the output.
+
+
+        Parameters
+        ----------
+        out_lons: array, :class:`numpy.ndarray`
```

Comment:
`pyproj` isn't strictly tied to `numpy`. It does support `numpy`, but it also supports quite a variety of other types as well. Initializing a python array within the function and returning it allows for this to be useful to a wider range of users and simplifies usage. Users can always convert the output to a numpy array if desired.

Reply:
The reason behind this choice was to make it as efficient as possible so the user can reuse this buffer for successive calls.
I will try to use a solution that would address both use cases.